File: discord_bot.js
const Logger = require("@ptkdev/logger");
const { Client, GatewayIntentBits, ActivityType, PresenceUpdateStatus } = require('discord.js');
const { status } = require('minecraft-server-util');
const readline = require('readline');
const fs = require('fs').promises;
const path = require('path');

const logger = new Logger();

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

// Promise-based question function
const question = (query) => new Promise((resolve) => {
    logger.info(`Prompting: ${query}`);
    rl.question(query, (answer) => { logger.warning(`Received input for: ${query}`); resolve(answer);});
});

async function setupConfig() {
    try {
        logger.docs("Welcome to Minecraft Server Status Bot Setup!", "Powered By @PPekKunGzDev");
        
        const token = await question("#Enter your Discord bot token > ");
        const ip = await question("#Enter your Minecraft server IP > ");
        const port = await question("#Enter your Minecraft server port (default: 25565) > ");

        const config = {
            TOKEN: token,
            IP: ip,
            PORT: parseInt(port) || 25565
        };
        
        await fs.writeFile(
            path.join(__dirname, 'config.json'), 
            JSON.stringify(config, null, 2)
        );
        
        logger.docs("Configuration saved successfully!", "Powered By @PPekKunGzDev");
        return config;
    } catch (error) {
        logger.error("Error during setup:", error);
        throw error;
    } finally {
        rl.close();
    }
}

async function loadConfig() {
    try {
        const configData = await fs.readFile(path.join(__dirname, 'config.json'), 'utf8');
        return JSON.parse(configData);
    } catch (error) {
        logger.info("No config file found. Starting setup...");
        return await setupConfig();
    }
}

async function startBot(config) {
    const client = new Client({
        intents: [GatewayIntentBits.Guilds],
    });

    client.once('ready', () => {
        logger.info("This bot is ready to go! | @PPekKunGzDev");
        logger.info(`Logged in as ${client.user.tag}`);
        checkServerStatus(client, config);
        setInterval(() => checkServerStatus(client, config), 60000);
    });

    async function checkServerStatus(client, config) {
        try {
            const serverStatus = await status(config.IP, config.PORT);
            
            client.user.setPresence({
                status: PresenceUpdateStatus.Online,
                activities: [{
                    name: `${serverStatus.players.online}/${serverStatus.players.max} players online`,
                    type: ActivityType.Watching
                }],
            });
            logger.info(`Server is online: ${serverStatus.players.online}/${serverStatus.players.max} players`);
        } catch (error) {
            logger.error("Server is Offline | @PPekKunGzDev");
            client.user.setPresence({
                status: PresenceUpdateStatus.Idle,
                activities: [{
                    url: 'https://dimension-studio.net',
                    name: 'Server is Offline',
                    type: ActivityType.Streaming
                }],
            });
        }
    }

    client.login(config.TOKEN);
}

async function main() {
    try {
        const config = await loadConfig();
        await startBot(config);
    } catch (error) {
        logger.error("Failed to start bot:", error);
        process.exit(1);
    }
}

main();

File: discord_bot.py
import discord
from discord.ext import tasks
from mcstatus import MinecraftServer
from rich.console import Console
from rich.prompt import Prompt
from rich.text import Text
from rich.panel import Panel
import json
import asyncio
import os

# Initialize rich console
console = Console()

async def setup_config():
    """Setup configuration for the bot with a fancy CLI."""
    try:
        console.print(Panel("Welcome to [bold green]Minecraft Server Status Bot Setup![/bold green]", title="Setup"))
        
        token = Prompt.ask("[cyan]Enter your Discord bot token[/cyan]")
        ip = Prompt.ask("[cyan]Enter your Minecraft server IP[/cyan]")
        port = Prompt.ask("[cyan]Enter your Minecraft server port (default: 25565)[/cyan]", default="25565")

        config = {
            "TOKEN": token.strip(),
            "IP": ip.strip(),
            "PORT": int(port.strip())
        }

        with open("config.json", "w") as config_file:
            json.dump(config, config_file, indent=2)
        
        console.print(Panel("[bold green]Configuration saved successfully![/bold green]", title="Success"))
        return config
    except Exception as e:
        console.print(Panel(f"[bold red]Error during setup: {e}[/bold red]", title="Error"))
        raise

async def load_config():
    """Load configuration from the file."""
    if not os.path.exists("config.json"):
        console.print(Panel("[yellow]No config file found. Starting setup...[/yellow]", title="Info"))
        return await setup_config()
    
    with open("config.json", "r") as config_file:
        return json.load(config_file)

class MinecraftStatusBot(discord.Client):
    def __init__(self, config, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.config = config
        self.server = MinecraftServer.lookup(f"{config['IP']}:{config['PORT']}")
        self.check_server_status.start()

    async def on_ready(self):
        console.print(Panel(f"Bot is ready! Logged in as [bold cyan]{self.user}[/bold cyan]", title="Bot Status"))

    @tasks.loop(minutes=1)
    async def check_server_status(self):
        try:
            status = self.server.status()
            activity = discord.Game(f"{status.players.online}/{status.players.max} players online")
            await self.change_presence(status=discord.Status.online, activity=activity)
            console.print(f"[green]Server is online:[/green] {status.players.online}/{status.players.max} players")
        except Exception as e:
            console.print("[red]Server is Offline[/red]")
            activity = discord.Streaming(name="Server is Offline", url="https://dimension-studio.net")
            await self.change_presence(status=discord.Status.idle, activity=activity)

    @check_server_status.before_loop
    async def before_check_server_status(self):
        await self.wait_until_ready()

async def main():
    try:
        config = await load_config()
        intents = discord.Intents.default()
        bot = MinecraftStatusBot(config, intents=intents)
        await bot.start(config["TOKEN"])
    except Exception as e:
        console.print(Panel(f"[bold red]Failed to start bot: {e}[/bold red]", title="Error"))
        exit(1)

if __name__ == "__main__":
    asyncio.run(main())
